src/Bicep.Core/Emit/ExpressionEmitter.cs

```diff
@@ -314,6 +314,46 @@ public void EmitObjectProperties(ObjectSyntax objectSyntax, ISet<string>? proper
             }
         }
 
+        public void EmitModuleParameterValue(SyntaxBase syntax)
+        {
+            if (syntax is InstanceFunctionCallSyntax instanceFunctionCall && string.Equals(instanceFunctionCall.Name.IdentifierName, "getSecret", StringComparison.Ordinal))
```

Comment:
>string.Equals(instanceFunctionCall.Name.IdentifierName, "getSecret" [](http://example.com/codeflow?start=77&length=67)

@anthony-c-martin Don't we have a helper somewhere that checks the bound symbol to verify that the function is the function we think it is?

Reply:
My main concern is that in the future we could have a `getSecret` function in one of the imported namespaces that behaves differently to the method on a KV.

Reply:
I can't think of anything immediately. The next best thing would be to type check the base expression on the InstanceFunctionCallSyntax and verify it is a resource type, and also that it's the particular resource type you expect (KV).

Reply:
This is being checked. You function getSecret is only available on keyVault type.

Reply:
I’d need to add to test for it.

Reply:
> This is being checked. Function getSecret is only available on keyVault type.

It is today, but I think Marcin's concern is that if we're not specific now, we're going to potentially generate invalid code if we ever add a function named `getSecret` to any other object type.

Reply:
true, I'll add a check then. I had it before AFAIR,  but got lost when refactoring :)

Reply:
I’ve an idea how to improve it.

what if instead writing emit instructions here I’d modify code to call a lambda that is associated with the type that is going to be assigned to parameter? I did something similar in includeContent function (still in progress though). The lambda would be defined with the function builder. WDYT?